lib/app/service/vendor.go

```diff
@@ -701,13 +716,22 @@ func isChartDirectory(path string) (bool, error) {
 	return !fi.IsDir(), nil
 }
 
-func resourceFromChart(path string) (*resources.Resource, error) {
-	out, err := helm.Render(helm.RenderParameters{Path: path})
+func resourcesFromChart(path string) (resources.ResourceFiles, error) {
+	rendered, err := helm.Render(helm.RenderParameters{Path: path})
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	log.WithField("path", path).Debug(string(out))
-	return resources.Decode(bytes.NewReader(out))
+	var result resources.ResourceFiles
+	for k, v := range rendered {
+		resource, err := resources.Decode(strings.NewReader(v))
+		if err != nil {
+			return nil, trace.Wrap(err).
+				AddUserMessage("%v", err).
```

Comment:
hmm, this is a weird use of the user message. Isn't this what `Wrap` is supposed to preserve in the first place?

Reply:
No, actually - the way it works now is Wrap does not add a user message unless it was provided explicitly like `trace.Wrap(err, "user message", ...)`. We could change that to extract a user message from the error although I think it might mess up all our current error reporting. Or I could just do `trace.Wrap(err, "%v", err)` here which would have the same effect.